### ISSUE TITLE
Use DotNetCoreNuGet instead of NuGet when running .NET Core builds

### ISSUE THREAD

**Issue by AdmiringWorm** [cake-contrib/Cake.Recipe#569]
When running .NET Core build we should push and pack packages using the dotnet utility instead of using nuget directly.

This would set us up for the future when we are able to run Cake.Recipe with cake .NET Core without the need to have users installing NuGet just for pushing/packing packages.

Thoughts @gep13

**Comment by gep13** [comment]
@AdmiringWorm I think this makes a lot of sense.  My only concern being the ability to also create the required sources that we currently do when using NuGet.

**Comment by AdmiringWorm** [comment]
@gep13 you thinking of the usage of `NuGetAddSource`?
If so, then there is an equivalent called `DotNetCoreNuGetAddSource`

**Comment by gep13** [comment]
Yes, that was what I was thinking about.  Never used `DotNetCoreNuGetAddSource` and I have no reason to think that it won't work, just need to test this out.

**Comment by AdmiringWorm** [comment]
This one will get pushed back until the time Cake.Recipe gets updated to use cake 0.38.0.

Seems that `DotNetCoreNuGetAddSource` was added in that version, which I would need to make pushing with dotnet equal to pushing with nuget.

**Comment by gep13** [comment]
@AdmiringWorm that is a good catch!
